src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -413,6 +413,7 @@ await foreach (var response in requests)
             ProvideSemanticTokensRangeParams semanticTokensParams,
             CancellationToken cancellationToken)
         {
+            // We're not actually serializing the Range correctly here. Likely we need to define a jsonconverter for MS.VS.Range
```

Comment:
This seems worrying. Was this true for when it was an O# range?

Reply:
As above, I left this as a reminder to myself and forgot to remove it. (For those interested, the solution was to add `DataMember` attributes to the range item to ensure serialization now that it's not a record.)